### PR TITLE
changing api to deploy - breaking change

### PR DESCRIPTION
**Pull request by strekm**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: parallel-install/pkg/components/component.go [hunk 1]
```diff
@@ -12,20 +12,20 @@ const StatusUninstalled = "Uninstalled"
 
 const logPrefix = "[components/component.go]"
 
-//ComponentInstallation interface defines a contract for Component installation and uninstallation.
-type ComponentInstallation interface {
-	//InstallComponent installs a component.
+//ComponentDeployment interface defines a contract for Component installation and uninstallation.
```
Comment (Tomasz-Smelcerz-SAP) [inline parallel-install/pkg/components/component.go:15]:
```suggestion
//ComponentDeploy interface defines a contract for Component installation and uninstallation.
```

File: parallel-install/pkg/components/component.go [hunk 2]
```diff
@@ -12,20 +12,20 @@ const StatusUninstalled = "Uninstalled"
...
+	DeployComponent(context.Context) error
 
 	//UninstallComponent uninstalls a component.
 	//The function is blocking until the component is uninstalled or an error (including Helm timeout) occurs.
 	//See the helm.HelmClient.UninstallRelease documentation for how context.Context is used for cancellation.
 	UninstallComponent(context.Context) error
 }
 
-//Component implements the ComponentInstallation interface.
+//Component implements the ComponentDeployment interface.
```
Comment (Tomasz-Smelcerz-SAP) [inline parallel-install/pkg/components/component.go:28]:
```suggestion
//Component implements the ComponentDeploy interface.
```

File: parallel-install/pkg/components/component.go [hunk 3]
```diff
@@ -54,13 +54,13 @@ func NewComponent(name, namespace, chartDir string, overrides func() map[string]
 	}
 }
 
-//InstallComponent implements ComponentInstallation.InstallComponent.
-func (c *Component) InstallComponent(ctx context.Context) error {
+//DeployComponent implements ComponentDeployment.DeployComponent.
```
Comment (Tomasz-Smelcerz-SAP) [inline parallel-install/pkg/components/component.go:57]: This looks so redundant now... I would remove this doc entirely.

File: parallel-install/pkg/engine/engine.go [hunk 4]
```diff
@@ -163,8 +163,8 @@ func worker(ctx context.Context, wg *sync.WaitGroup, jobChan <-chan components.C
 				return
 			}
 			if ok {
-				if installationType == "install" {
-					if err := component.InstallComponent(ctx); err != nil {
+				if installationType == "deploy" {
+					if err := component.DeployComponent(ctx); err != nil {
 						component.Status = components.StatusError
 					} else {
 						component.Status = components.StatusInstalled
```
Comment (Tomasz-Smelcerz-SAP) [inline parallel-install/pkg/engine/engine.go:170]: I would really consider adding another status here: StatusUpgraded. This provides feedback to library users what actually happened (install vs upgrade)

Comment (strekm) [reply parallel-install/pkg/engine/engine.go:170]: we can think about it in next PR implementing upgrades

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**